quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/ApplicationProfileConfigSource.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.quarkus.providers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.lsp4mp.jdt.core.project.PropertiesConfigSource;
+
+/**
+ * Represents <code>application-${profile}.properties</code> config sources.
+ *
+ * This config source contributes configuration only for a particular profile.
+ * It takes precedence over <code>application.properties</code>.
+ *
+ */
+public class ApplicationProfileConfigSource extends PropertiesConfigSource {
+
+	public static final Pattern FILE_NAME_PTN = Pattern.compile("application-([A-Za-z]+)\\.properties");
+
+	private final String profile;
+
+	public ApplicationProfileConfigSource(String configFileName, IJavaProject javaProject) {
+		// I don't think Quarkus assigns a specific ordinal to
+		// application-${profile}.properties files.
+		// This ordinal means that application-${profile}.properties files take
+		// precedence over application.properties
+		super(configFileName, javaProject, 261);
+		Matcher m = FILE_NAME_PTN.matcher(configFileName);
+		if (!m.matches()) {
```

Comment:
Why you check that?

Reply:
I was just using this to make sure that making an invalid ApplicationProfileConfigSource would lead to an exception. I can remove it.

Reply:
When you create an instance of ApplicationProfileConfigSource  you are usre that you check that before, so don't need to double check it.

Reply:
Okay